autometa/common/external/samtools.py

```diff
@@ -26,58 +26,65 @@
 import logging
 import os
 import subprocess
+import shutil
+import tempfile
 
 import pandas as pd
 import multiprocessing as mp
 
 logger = logging.getLogger(__name__)
 
 
-def run(cmd):
-    """Run `cmd` via subprocess.
-
+def run(cmd, bam):
+    """
+    Run `cmd` via subprocess.
+    
     Parameters
     ----------
     cmd : str
         Executable input str
-
+    bam : str
+        </path/to/output/alignment.bam>
+    
     Returns
     -------
     bool
-        True if retcode == 0. i.e. Success.
-        False if retcode != 0. i.e. Failed.
-
+        True if retcode == 0. i.e. function run successfully.
+    
     Raises
-    -------
-    ExceptionName
-        Why the exception is raised.
-
+    ------
+    ChildProcessError
+        Function did not run successfully, returns retcode.
     """
+    
     logger.debug(f'cmd: {cmd}')
-    with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
+    log_samtools_dir = os.path.dirname(bam)
+    tempdir = tempfile.mkdtemp(suffix=None, prefix='samtools', dir=log_samtools_dir)
+    samtools_stderr=os.path.join(tempdir, 'samtools_stderr')
+    samtools_stdout=os.path.join(tempdir, 'samtools_stdout')
```

Comment:
Typically stderr and stdout are denoted as `.err` and `.out` respectively. If you like, you could also keep the stderr and stdout but add them as extensions to the samtools file name. I.e. `samtools.stderr` or `samtools.err` and `samtools.stdout` or `samtools.out`